pkg/api/store/cluster/cluster_store.go

```diff
@@ -17,3 +26,29 @@ func (r *Store) ByID(apiContext *types.APIContext, schema *types.Schema, id stri
 	}
 	return r.Store.ByID(apiContext, schema, id)
 }
+
+func (r *Store) Create(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}) (map[string]interface{}, error) {
+
+	name, ok := data["name"].(string)
+	if !ok {
+		return nil, errors.New("Cluster name is required")
```

Comment:
@prachidamle do we need this extra check? Given that required attribute is already specified as norman annotation

Reply:
@alena1108 yeah we need this check since the API request data is sent in a map and not as the Cluster object, so need to check the field to avoid panics while casting to string